File: src/common/Smi.Common/Messages/Extraction/ExtractFileStatus.cs

namespace Smi.Common.Messages.Extraction
{
    // TODO(rkm 2020-03-07) Check what errors CTPAnonymiser can actually spit out here
    public enum ExtractFileStatus
    {
        Unknown = 0,

        /// <summary>
        /// The file has been anonymised successfully
        /// </summary>
        Anonymised,

        /// <summary>
        /// The file could not be anonymised but will be retried later
        /// </summary>
        ErrorWillRetry,

        /// <summary>
        /// The file could not be anonymised and will not be retired
        /// </summary>
        ErrorWontRetry,
        
        // TODO Java
        /// <summary>
        /// The source file could not be found under the given filesystem root
        /// </summary>
        FileMissing,

        // TODO Java
        /// <summary>
        /// The source file was successfully copied to the destination
        /// </summary>
        Copied,
    }
}

File: src/microservices/com.smi.microservices.ctpanonymiser/src/main/java/org/smi/ctpanonymiser/util/ExtractFileStatus.java

package org.smi.ctpanonymiser.util;

public enum ExtractFileStatus {

    Unknown,

	/**
	 * The file has been anonymised successfully
	 */
	Anonymised,

	/**
	 * The file could not be anonymised but will be retried later
	 */
	ErrorWillRetry,

	/**
	 * The file could not be anonymised and will not be retired
	 */
	ErrorWontRetry,

	/**
	 * The source file could not be found under the given filesystem root
	 */
	FileMissing,

	/**
	 * The source file was successfully copied to the destination
	 */
	Copied,
}
